ee/tabby-db/src/integration_access_token.rs

```diff
@@ -0,0 +1,139 @@
+use anyhow::{anyhow, Result};
+use sqlx::{prelude::FromRow, query, query_as};
+use tabby_db_macros::query_paged_as;
+
+use crate::{DateTimeUtc, DbConn};
+
+#[derive(FromRow)]
+pub struct IntegrationAccessTokenDAO {
+    pub id: i64,
+    pub kind: String,
+    pub display_name: String,
+    pub access_token: Option<String>,
+    pub created_at: DateTimeUtc,
+    pub updated_at: DateTimeUtc,
+}
+
+impl DbConn {
+    pub async fn create_integration_access_token(
+        &self,
+        kind: &str,
+        name: &str,
+        access_token: &str,
+    ) -> Result<i64> {
+        let res = query!(
+            "INSERT INTO integration_access_tokens(kind, display_name, access_token) VALUES (?, ?, ?);",
+            kind,
+            name,
+            access_token
+        )
+        .execute(&self.pool)
+        .await?;
+        Ok(res.last_insert_rowid())
+    }
+
+    pub async fn get_integration_access_token(&self, id: i64) -> Result<IntegrationAccessTokenDAO> {
+        let provider = query_as!(
+            IntegrationAccessTokenDAO,
+            r#"SELECT
+                id,
+                kind,
+                display_name,
+                access_token,
+                created_at AS "created_at: DateTimeUtc",
+                updated_at AS "updated_at: DateTimeUtc"
+            FROM integration_access_tokens WHERE id = ?;"#,
+            id
+        )
+        .fetch_one(&self.pool)
+        .await?;
+        Ok(provider)
+    }
+
+    pub async fn delete_integration_access_token(&self, id: i64) -> Result<()> {
+        let res = query!("DELETE FROM integration_access_tokens WHERE id = ?;", id)
+            .execute(&self.pool)
+            .await?;
+        if res.rows_affected() != 1 {
+            return Err(anyhow!("No integration access token to delete"));
+        }
+        Ok(())
+    }
+
+    pub async fn update_integration_access_token(
+        &self,
+        id: i64,
+        display_name: String,
+        access_token: String,
+    ) -> Result<()> {
+        let res = query!(
+            "UPDATE integration_access_tokens SET display_name = ?, access_token=? WHERE id = ?;",
+            display_name,
+            access_token,
+            id
+        )
+        .execute(&self.pool)
+        .await?;
+
+        if res.rows_affected() != 1 {
+            return Err(anyhow!(
+                "The specified integration access token does not exist"
+            ));
+        }
+
+        Ok(())
+    }
+
+    pub async fn update_integration_access_token_sync_status(
```

Comment:
```suggestion
    pub async fn update_integration_access_token_valid(
```

Reply:
Done, but since `updated_at` is non-nullable where `synced_at` was nullable before, we are only representing 2 possible states with `valid` now. How do we want to represent ready vs pending vs error?

Reply:
Good point, let's use a field `error: Option<String>` with a derived DAO method `valid = error == null`

Reply:
I'll leave `valid` to be converted in the service layer

Reply:
If `created_at` == `synced_at` and `error` is null, then the status will be `pending`. Otherwise status is `ready` as long as `error` is null

Reply:
Come to think of it, this still works even if we revert to using a bool `valid`: `created_at == synced_at` -> pending, `valid && created_at != synced_at` -> ready, `!valid && created_at != synced_at` -> error

Reply:
let‘s add a `status` enum string field for more explicity.

Reply:
Let's still go with `error` approach for better explicity